src/App.tsx

```diff
@@ -1,24 +1,140 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { FilteredBy, filteredTodoList } from './helpers';
+import { Errors } from './components/Errors/Errors';
+import { ErrorType } from './types/ErorTypes';
 
-const USER_ID = 0;
+const USER_ID = 12036;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.DefaultType);
+  const [errorType, setErorType] = useState<ErrorType | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const todoToRemove = async (todoId: number) => {
+    try {
+      await todoService.removeTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch (error) {
+      setErorType(ErrorType.DeleteError);
+    }
+  };
+
+  const addTodo = async (todoTitle: string) => {
+    try {
+      setIsLoading(true);
+      const newTempTodo: Todo = {
+        id: 0,
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTempTodo(newTempTodo);
+
+      const addedTodo = await todoService.addTodo({
+        title: todoTitle,
+        completed: false,
+        userId: USER_ID,
+      });
+
+      if (addedTodo.id !== 0) {
+        setTempTodo({
+          ...newTempTodo,
+          id: addedTodo.id,
+        });
+      }
```

Comment:
Looks like this step is redundant, as when you have a real todo you just set it  to todos list